src/content/docs/en/guides/assets.mdx

```diff
@@ -0,0 +1,277 @@
+:::
+
+Import images from a **relative file path** or [import alias](/en/guides/aliases/) in any component file and then use the import as the `<Image />` component's `src` attribute.
+
+```astro
+---
+import { Image } from 'astro:assets';
+import myImage from "../assets/my_image.png"; // Image is 1600x900
+---
+
+<!-- `alt` is mandatory on the Image component -->
+<Image src={myImage} alt="A description of my image." />
+```
+
+```astro
+<!-- Output -->
+<!-- Image is optimized, proper attributes are enforced -->
+<img
+  src="/_astro/my_image.hash.webp"
+  width="1600"
+  height="900"
+  decoding="async"
+  loading="lazy"
+  alt="A description of my image."
+/>
+```
+
+#### Properties
+
+##### width and height
+
+These properties define the dimensions to use for the image.
+
+When using local images in their original aspect ratio, the `width` and `height` can be automatically inferred from the source file and are optional. However, both of these properties are required for remote images.
+
+##### format
+
+You can optionally state the [image file type](https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types#common_image_file_types) to be used. The default file format used is `webp`.
+
+##### quality
+
+`quality` is an optional property that can either be:
+- a preset (`low`, `medium`, `high`, `max`) that is automatically normalized between formats.
+- a number from `0` to `100` (interpreted differently between formats).
+
+##### alt (required)
+
+Use the `alt` attribute to provide [descriptive alt text](https://www.w3.org/WAI/tutorials/images/) for images.
+
+This attribute is required for the `<Image />` component. This component will throw an error if no alt text is provided.
+
+If the image is merely decorative (i.e. doesn't contribute to the understanding of the page), set `alt=""` so that screen readers know to ignore the image.
+
+##### Additional properties
+
+In addition to the properties above, the `<Image />` component accepts all properties accepted by the HTML `<img>` tag. 
+
+For example, you can provide a `class` to the final `img` element.
+
+```astro
+---
+import { Image } from 'astro:assets';
+import myImage from "../assets/my_image.png";
+---
+
+<!-- `alt` is mandatory on the Image component -->
+<Image src={myImage} alt="" class="my-class" />
+```
+
+```astro
+<!-- Output -->
+<img
+  src="/_astro/my_image.hash.webp"
+  width="1600"
+  height="900"
+  decoding="async"
+  loading="lazy"
+  class="my-class"
+  alt=""
+/>
+```
+
+### `getImage()` (`astro:assets`)
+
+This function is intended for generating images destined to be used somewhere else than directly in HTML, for example in CSS or in an [API Route](/en/core-concepts/endpoints/#server-endpoints-api-routes).
```

Comment:
A major usage of `getImage` is also if you want to create your own `Image` component, might be worth a shoutout (the included Image component is basically just a call to `getImage`!)